### PR TITLE
[DNMY] New NUT configurator feature image

### PR DESCRIPTION
**Pull request by eric3873**
Part of the feature image "new-nut-configurator".

### CODE REVIEW

File: src/fty_nut_configuration_server.cc [hunk 1]
```diff
@@ -0,0 +1,848 @@
...
+                        break;
+                    }
+                    it ++;
+                }
+                if (find) {
+                    detectedFingerprintIterator = it_detectedFingerprints;
+                    break;
+                }
+                it_detectedFingerprints ++;
+            }
```
Comment (boricj) [inline src/fty_nut_configuration_server.cc]: Check if replaceable with `std::find`.

File: include/fty_nut_configuration_manager.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        ConfigurationManager(const std::string& dbConn);
+        ~ConfigurationManager() = default;
+
+        static std::string serialize_config(std::string name, nutcommon::DeviceConfiguration& config);
+        void automaticAssetConfigurationPrioritySort(fty_proto_t* asset);
+        void scanAssetConfigurations(fty_proto_t* asset);
+
+        void getAssetConfigurations(fty_proto_t* asset, nutcommon::DeviceConfigurations& configs);
+        void getAssetConfigurationsWithSecwDocuments(fty_proto_t* asset, nutcommon::DeviceConfigurations& configs, std::set<secw::Id>& secw_document_id_list);
+        void saveAssetConfigurations(std::string asset_name, nutcommon::DeviceConfigurations& configs, std::set<secw::Id>& secw_document_id_list);
```
Comment (boricj) [inline include/fty_nut_configuration_manager.h]: Const-correctness for `std::string`.

File: src/fty_nut_command_server.cc [hunk 3]
```diff
@@ -561,14 +561,14 @@ NutCommandConnector::NutCommandConnector(NutCommandConnector::Parameters params)
...
-        (msg.metaData().count(messagebus::Message::COORELATION_ID) == 0) ||
+        (msg.metaData().count(messagebus::Message::CORRELATION_ID) == 0) ||
         (msg.metaData().count(messagebus::Message::REPLY_TO) == 0)) {
         log_error("Missing subject/correlationID/replyTo in request.");
         return;
     }
 
     auto subject = msg.metaData()[messagebus::Message::SUBJECT];
-    auto corrId = msg.metaData()[messagebus::Message::COORELATION_ID];
+    auto corrId = msg.metaData()[messagebus::Message::CORRELATION_ID];
```
Comment (boricj) [inline src/fty_nut_command_server.cc]: Check if corrected in master and possibly rebase on top of master.

**Comment by perrettecl** [comment]
Not used for now

**Pull request closed without merging**